File: 4_gemini_sol2.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        // Initialize fast I/O readers
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        String line = br.readLine();
        if (line == null) return;
        
        StringTokenizer st = new StringTokenizer(line);
        int n = Integer.parseInt(st.nextToken());
        
        st = new StringTokenizer(br.readLine());
        long totalMilk = 0;
        
        for (int i = 0; i < n; i++) {
            totalMilk += Long.parseLong(st.nextToken());
        }
        
        // Ternary operator for concise output
        System.out.println(totalMilk % n == 0 ? "YES" : "NO");
    }
}

File: 4_gemini_sol2.cpp
#include <iostream>
#include <vector>
#include <numeric>

using namespace std;

int main() {
    int n;
    if (cin >> n) {
        vector<long long> milk_glasses(n);
        
        // Read all inputs into the vector
        for (int i = 0; i < n; i++) {
            cin >> milk_glasses[i];
        }

        // 0LL ensures the sum is accumulated as a 64-bit integer
        long long total_milk = accumulate(milk_glasses.begin(), milk_glasses.end(), 0LL);

        // Ternary operator for concise output
        cout << (total_milk % n == 0 ? "YES\n" : "NO\n");
    }
    return 0;
}
